sdk/eventgrid/Microsoft.Azure.WebJobs.Extensions.EventGrid/src/EventGridExtensionConfigProvider.cs

```diff
@@ -84,6 +84,7 @@ public void Initialize(ExtensionConfigContext context)
                 .AddBindingRule<EventGridTriggerAttribute>() // following converters are for EventGridTriggerAttribute only
                 .AddConverter<JToken, string>(jtoken => jtoken.ToString(Formatting.Indented))
                 .AddConverter<JToken, string[]>(jarray => jarray.Select(ar => ar.ToString(Formatting.Indented)).ToArray())
+                .AddConverter<DirectInvokeString, JToken>((directInvokeString) => JToken.Parse(directInvokeString.Value))
```

Comment:
Can you clarify how we can test this in the portal?

Reply:
the issue with repro steps:
https://github.com/Azure/azure-functions-host/issues/6328

a unit test is not possible as portal test UI uses admin function host webapi

Reply:
I don't see why a unit test wouldn't be possible here. Couldn't we attempt to deliver a DirectInvokeString to a function that binds using a JToken?
I tried this locally, and I get an exception here - https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/eventgrid/Microsoft.Azure.WebJobs.Extensions.EventGrid/src/TriggerBinding/EventGridTriggerAttributeBindingProvider.cs#L129

Do we need an update to the binding provider as well?

Reply:
Also, if this worked prior to https://github.com/Azure/azure-functions-eventgrid-extension/commit/a12eb6d5749f98dc0c9726e0ae6c7b4a79eaca72, shouldn't we be reverting the converters to that rather than adding different converters? It makes me nervous to have new code that we aren't able to test as opposed to reverting to a known correct version.

Reply:
I ended up with a safer fix - a new convertor added, all old convertors remain the same

Reply:
Thanks for doing that!